win32/tests/job_object_helper_ut/job_object_helper_ut.c

```diff
@@ -0,0 +1,386 @@
+// Copyright(C) Microsoft Corporation.All rights reserved.
+
+
+#include <stdlib.h>
+
+#include "windows.h"
+
+#include "macro_utils/macro_utils.h"
+
+#include "testrunnerswitcher.h"
+#include "umock_c/umock_c.h"
+#include "umock_c/umock_c_negative_tests.h"
+#include "umock_c/umocktypes_windows.h"
+
+#define ENABLE_MOCKS
+    #include "c_pal/interlocked.h"
+    #include "c_pal/interlocked_hl.h"
+    #include "c_pal/gballoc_hl.h"
+    #include "c_pal/gballoc_hl_redirect.h"
+
+    MOCKABLE_FUNCTION(, HANDLE, mocked_CreateJobObject, LPSECURITY_ATTRIBUTES, lpJobAttributes, LPCSTR, lpName);
+    MOCKABLE_FUNCTION(, HANDLE, mocked_GetCurrentProcess)
+    MOCKABLE_FUNCTION(, BOOL, mocked_AssignProcessToJobObject, HANDLE, hJob, HANDLE, hProcess);
+    MOCKABLE_FUNCTION(, BOOL, mocked_CloseHandle, HANDLE, hObject);
+
+#undef ENABLE_MOCKS
+
+#include "real_gballoc_hl.h"
+#include "c_pal/job_object_helper.h"
+
+MU_DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+{
+    ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
+}
+
+#define test_percent_cpu 42
+#define test_percent_physical_memory 89
+#define test_job_object (HANDLE)0x42001
+#define test_process_handle (HANDLE)0x42002
+#define test_total_physical_memory 20480
+
+MOCK_FUNCTION_WITH_CODE(, BOOL, mocked_GlobalMemoryStatusEx, LPMEMORYSTATUSEX, lpBuffer)
+{
+    ASSERT_IS_NOT_NULL(lpBuffer);
+    ASSERT_ARE_EQUAL(size_t, lpBuffer->dwLength, sizeof(MEMORYSTATUSEX));
+    (void)memset(lpBuffer, 0, sizeof(MEMORYSTATUSEX));
+    lpBuffer->dwLength = sizeof(MEMORYSTATUSEX);
+    lpBuffer->ullTotalPhys = test_total_physical_memory;
+}
+MOCK_FUNCTION_END(TRUE)
+
+static JOBOBJECT_EXTENDED_LIMIT_INFORMATION captured_extended_limit_information;
+static JOBOBJECT_CPU_RATE_CONTROL_INFORMATION captured_cpu_rate_control_information;
+MOCK_FUNCTION_WITH_CODE(, BOOL, mocked_SetInformationJobObject, HANDLE, hJob, JOBOBJECTINFOCLASS, JobObjectInformationClass, LPVOID, lpJobObjectInformation, DWORD, cbJobObjectInformationLength)
+{
+    ASSERT_IS_NOT_NULL(lpJobObjectInformation);
+    if (JobObjectInformationClass == JobObjectExtendedLimitInformation)
+    {
+        (void)memcpy(&captured_extended_limit_information, lpJobObjectInformation, cbJobObjectInformationLength);
```

Comment:
>(void)memcpy(&captured_extended_limit_information, lpJobObjectInformation, cbJobObjectInformationLength); [](http://example.com/codeflow?start=8&length=105)

why is captured_extended_limit_information variable needed?

Reply:
same for captured_cpu_rate_control_information

Reply:
I cannot find them being accessed anywhere except here.

Reply:
I forgot to add the `ASSERT`s that use these values.  Adding them now.